docs/src/developers/style.md

```diff
@@ -51,6 +51,162 @@ format("test")
 The following sections outline extra style guide points that are not fixed
 automatically by JuliaFormatter.
 
+### [The `MethodError` principle](@id method_error_principle)
```

Comment:
I think this can be rewritten more clearly as a corollary of the guidelines for dealing the abstraction and composition, rather than a standalone principle.